src/ray/util/container_util.h

```diff
@@ -28,56 +30,92 @@
 namespace ray {
 
 template <typename T>
-std::string debug_string(const T &obj) {
-  std::stringstream ss;
-  ss << obj;
-  return ss.str();
+class DebugStringWrapper;
+
+// The actual interface.
+template <typename T>
+DebugStringWrapper<T> debug_string(const T &t) {
+  return DebugStringWrapper<T>(t);
+}
+
+/// Wrapper for `debug_string(const T&)`.
+template <typename T>
+class DebugStringWrapper {
+ public:
+  explicit DebugStringWrapper(const T &obj) : obj_(obj) {}
+
+  // Only initialized for the blessed container types listed below with operator<<
+  // specializations.
+  std::ostream &_container_debug_string_impl(std::ostream &os) const {
```

Comment:
Curious what's the naming convension?
Google coding style encourages `SnakeCase`, ref: https://google.github.io/styleguide/cppguide.html#Function_Names

Reply:
done